### DISCUSSION TITLE
Bulk update or create

### DISCUSSION THREAD

**Discussion by yedidyaco** [collerek/ormar#802] [Q&A]
Hi,

I have the following scenario:

```python
class DataModel(ormar.Model):
     ...
     name: str = String(max_length=32, primary_key=True)
     amount: float = Float()
     ...
```

I'm getting a list of `DataModel` objects, and want to store them to the DB.
Some of the `DataModel` names are already in the DB and have an updated `amount` in them, some are entirely new - the `name` is not in the DB yet.

Is there any efficient way to make "bulk_update_or_create" in this scenario?

Thanks!

**Comment by ponytailer** [comment]
Maybe the `insert into (name, amount) values("hello", 12) on conflict(name) do update set amount=1.0`.
But `ormar` was not supported yet.
